infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/VcsSshKeysProvisioner.java

```diff
@@ -118,29 +117,39 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       }
     }
 
+    doProvisionSshKeys(sshPairs, k8sEnv, identity.getWorkspaceId());
+
     StringBuilder sshConfigData = new StringBuilder();
     for (SshPair sshPair : sshPairs) {
-      doProvisionSshKey(sshPair, k8sEnv, identity.getWorkspaceId());
+
       sshConfigData.append(buildConfig(sshPair.getName()));
     }
 
     String sshConfigMapName = identity.getWorkspaceId() + SSH_CONFIG_MAP_NAME_SUFFIX;
     doProvisionSshConfig(sshConfigMapName, sshConfigData.toString(), k8sEnv);
   }
 
-  private void doProvisionSshKey(SshPair sshPair, KubernetesEnvironment k8sEnv, String wsId) {
-    if (isNullOrEmpty(sshPair.getName()) || isNullOrEmpty(sshPair.getPrivateKey())) {
-      return;
-    }
-    String validNameForSecret = getValidNameForSecret(sshPair.getName());
+  private void doProvisionSshKeys(
+      List<SshPairImpl> sshPairs, KubernetesEnvironment k8sEnv, String wsId) {
+
+    Map<String, String> data =
+        sshPairs
+            .stream()
+            .filter(
+                sshPair ->
+                    !isNullOrEmpty(sshPair.getName()) && !isNullOrEmpty(sshPair.getPrivateKey()))
```

Comment:
I believe that name must not be null or empty https://github.com/eclipse/che/blob/07263f1e30089689d71b057f747a44a29283e3c4/wsmaster/che-core-api-ssh-shared/src/main/java/org/eclipse/che/api/ssh/shared/model/SshPair.java#L25

And PrivateKey may be null, so the check is needed.

Reply:
There is a filter that checks only non nullable ssh keys, isn't it?